server/odc-service/src/main/java/com/oceanbase/odc/service/task/schedule/daemon/CheckRunningJob.java

```diff
@@ -54,51 +62,86 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
         TaskFrameworkProperties taskFrameworkProperties = getConfiguration().getTaskFrameworkProperties();
         int size = taskFrameworkProperties.getSingleFetchCheckHeartTimeoutJobRows();
         int heartTimeoutPeriod = taskFrameworkProperties.getJobHeartTimeoutSeconds();
+        handleGeneralHeartTimeoutJobs(size, heartTimeoutPeriod);
+        if (taskFrameworkProperties.getRunMode().isProcess()) {
+            handleNotLocalAndProcessIsNotExitsJobs(size, heartTimeoutPeriod);
+        }
+    }
+
+    private void handleGeneralHeartTimeoutJobs(int size, int heartTimeoutPeriod) {
         // find heart timeout job
         Page<JobEntity> jobs = getConfiguration().getTaskFrameworkService()
                 .findHeartTimeTimeoutJobs(heartTimeoutPeriod, 0, size);
-        jobs.forEach(this::handleJobRetryingOrCanceled);
-
-    }
-
-    private void handleJobRetryingOrCanceled(JobEntity a) {
-        getConfiguration().getTransactionManager().doInTransactionWithoutResult(() -> {
-            doHandleJobRetryingOrFailed(a);
-        });
-
+        for (JobEntity j : jobs) {
+            SilentExecutor.executeSafely("handleJobRetryingOrFailed", () -> {
+                handleJobRetryingOrFailed(j, a -> {
+                    try {
```

Comment:
why should we catch the exception here?
doesn't you catch this exception in SilentExecutor?

Reply:
JobException is Check Exception, so i change it to RuntimeException, so no need to throw exception

Reply:
done, fetch 10000 timeout in process mode and destroy process in BaseJobCaller